### PR TITLE
support last_over_time function

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <benye@amazon.com>

This pr adds support for `last_over_time` function. 
One thing different is that `last_over_time` requires the metric name to be kept in the final vector result. So I just simply pass the `parser.Call` to the `NewMatrixSelector` to check whether to drop the metric name or not.

### CODE REVIEW

**Review by fpetkovski — APPROVED** [review]
Thanks @yeya24 👍

File: physicalplan/scan/function.go [hunk 1]
```diff
@@ -70,6 +70,16 @@ func NewFunctionCall(f *parser.Function, selectRange time.Duration) (FunctionCal
 				Metric: labels,
 			}
 		}, nil
+	case "last_over_time":
+		return func(labels labels.Labels, points []promql.Point, stepTime time.Time) promql.Sample {
+			return promql.Sample{
+				Point: promql.Point{
+					T: stepTime.UnixMilli(),
+					V: points[len(points)-1].V,
```
Comment (fpetkovski) [inline physicalplan/scan/function.go:78]: Do we need to check for out of bounds here? Will `points` always be a slice with at least one element?

Comment (yeya24) [reply physicalplan/scan/function.go:78]: Good question, I think that's something we need to handle.

Comment (yeya24) [reply physicalplan/scan/function.go:78]: Umm, is it possible to have the series with labels only but no sample? I guess no so this should be safe. If no series then the func won't be called

Comment (fpetkovski) [reply physicalplan/scan/function.go:78]: Maybe it can happen if there are gaps in the values, but I also need to double check if this is a real problem.

**Pull request merged**